src/test/java/com/Model/ControllerTest/AdminTest.java

```diff
@@ -52,7 +52,7 @@ public void AdminLoginTest () throws Exception {
         this.userRepository.save(adminUser);
         RequestBuilder requestBuilder = formLogin().user("emailaddress","test@test.test").password("password","testpassword").loginProcessingUrl("/userlogin");
         this.mockMvc.perform(requestBuilder)
-                .andExpect(authenticated().withUsername("test@test.test"));
+                .andExpect(authenticated().withUsername("test@test.test").withRoles("ADMIN", "USER"));
```

Comment:
why is it "ADMIN" in some places and "ADMIN_ROLE" in others?

Reply:
When testing roles, test methods will automatically and "ROLE_" here.